autox/informer.go

```diff
@@ -3,154 +3,359 @@ package autox
+		uid  string
+	}
+
+	// chart is the Helm chart for this application
+	chart struct {
+		name    string
+		values  map[string]interface{}
+		version string
+	}
+}
+
+// the name of a release will depend on:
+//
+//	the name of the releaseSpec,
+//	the ID of the spec within the releaseSpec, and
+//	the set of (pruned) labels that triggers this release
+func getReleaseName(group string, releaseSpecID string, prunedLabels map[string]string) string {
+
+	// use labels relevant to autoX to create a random hash value
+	// this value will be appended as a suffix in the release name
+	var hasher maphash.Hash
+	// specGroupName and specID are always hashed
+	_, _ = hasher.WriteString(group)
+	_, _ = hasher.WriteString(releaseSpecID)
+
+	// hash app label
+	app := prunedLabels[appLabel]
+	_, _ = hasher.WriteString(app)
+
+	// hash version label
+	version := prunedLabels[versionLabel]
+	_, _ = hasher.WriteString(version)
+
+	// hash track label
+	track := prunedLabels[trackLabel]
+	_, _ = hasher.WriteString(track)
+
+	nonce := fmt.Sprintf("%05x", hasher.Sum64())
+	nonce = nonce[:5]
+	return fmt.Sprintf("autox-%s-%s-%s", group, releaseSpecID, nonce)
 }
 
-func installHelmReleases(resourceName string) {
-	doChartAction(resourceName, releaseAction)
+// installHelmReleases for a given spec group
+func installHelmReleases(prunedLabels map[string]string, namespace string, autoXConfig config) error {
+	return doChartAction(prunedLabels, releaseAction, namespace, autoXConfig)
 }
 
-func installHelmRelease(releaseName string, chart chart) {
-	// TODO: check if there is a preexisting Helm release
+// installHelmRelease for a given spec within a spec group
+var installHelmRelease = func(releaseName string, group string, releaseSpec releaseSpec, namespace string) error {
+	secretsClient := k8sClient.clientset.CoreV1().Secrets(namespace)
+
+	// TODO: what to put for ctx?
+	// get secret, based on autoX label
+	labelSelector := fmt.Sprintf("%s=%s", autoXLabel, group)
+	secretList, err := secretsClient.List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		log.Logger.Error("could not list secrets")
+		return err
+	}
+
+	// ensure that only one secret was found
+	if secretsLen := len(secretList.Items); secretsLen == 0 {
+		log.Logger.Error("expected secret with label selector:", labelSelector, "but none were found")
+		return err
+	} else if secretsLen > 1 {
+		log.Logger.Error("expected secret with label selector:", labelSelector, "but more than one were found")
+		return err
+	}
+	secret := secretList.Items[0]
+
+	values := applicationValues{
+		name:      releaseName,
+		namespace: namespace,
+
+		owner: struct {
+			name string
+			uid  string
+		}{
+			name: secret.Name,
+			uid:  string(secret.GetUID()),
+		},
+
+		chart: struct {
+			name    string
+			values  map[string]interface{}
+			version string
+		}{
+			name:    releaseSpec.Name,
+			values:  releaseSpec.Values,
+			version: releaseSpec.Version,
+		},
+	}
```

Comment:
I saw that we took `repoURL` out of release spec. We are still setting `repoURL: {{ .chart.url }}` in `application.tpl`. What value are we using for `url`? If we are hard coding it, are we doing it here or in the template?

Reply:
We can take it out of the template too. For now, this will be hard coded to Iter8 Helm hub within autox code.